pkg/k8s/cnp.go

```diff
@@ -367,21 +367,58 @@ func updateStatusByCapabilities(client clientset.Interface, capabilities k8svers
 		}
 
 		_, err = client.CiliumV2().CiliumNetworkPolicies(ns).Patch(name, k8sTypes.JSONPatchType, createStatusAndNodePatchJSON, "status")
+		// If the patch fails it means the "test" from the previous patch
+		// failed so we can safely replace the nodes in the CNP status.
 		if err != nil {
-			// If it fails it means the test from the previous patch failed
-			// so we can safely replace this node in the CNP status.
-			createStatusAndNodePatch := []JSONPatch{
-				{
-					OP:    "replace",
-					Path:  "/status/nodes/" + nodeName,
-					Value: cnpns,
-				},
-			}
-			createStatusAndNodePatchJSON, err = json.Marshal(createStatusAndNodePatch)
-			if err != nil {
-				return err
+			// If there are more than MaxJSONPatchOperations to patch, do
```

Comment:
My motivation for doing it this way was to only have us replace the node status updates for the nodes which we have received events for. We would have to maintain the state of all NodeStatuses for the given CNP (which we already have in the map of node name --> NodeStatus passed into this function) and put that entire map into the PATCH every time we update it, right? My intuition was that this would lead to us sending unnecessary information to the K8s apiserver if say there was only one node that was updated, as we would only transmit the patch for that node, not all prior nodes + this new node. But, it would reduce the total number of patches that we would be transmitting. I don't have any hard data to back this up, however. I can try measuring both? What do you think? I am open to all suggestions here, and you have a lot of prior experience with optimizing K8s apiserver interactions.

Reply:
Newest set of commits is using MergePatch instead of just Patch, still WIP though. Need to run in case where offload is enabled vs. not enabled manually on my local setup.